### PR TITLE
Compare visually similar geometries

### PR DESCRIPTION
**Pull request by janper**
Check if two mesh geometries are visually similar. They can differ in vertex and normal order and the start of face winding.

### CODE REVIEW

File: src/mesh_analysis.rs [hunk 1]
```diff
@@ -144,6 +147,78 @@ pub fn triangulated_mesh_genus(vertex_count: usize, edge_count: usize, face_coun
     1 - (cast_i32(vertex_count) - cast_i32(edge_count) + cast_i32(face_count)) / 2
 }
 
+/// Checks if two geometries are visually identical.
+///
+/// The two geometries can
+/// possibly contain the same vertices and normals bound to the same faces, but all
+/// of these can be in different order, including shifts in face winding. This
```
Comment (yanchith) [inline src/mesh_analysis.rs]: > ... including shifts in face winding.

I understood this comment that we consider geometries with opposite winding still identical, a long as they are consistent, but the code doesn't seem to support that. Or?

File: src/mesh_analysis.rs [hunk 2]
```diff
@@ -144,6 +147,78 @@ pub fn triangulated_mesh_genus(vertex_count: usize, edge_count: usize, face_coun
...
+                    && self.vertices.1 == other.vertices.0
+                    && self.vertices.2 == other.vertices.1
+                    && self.normals.0 == other.normals.2
+                    && self.normals.1 == other.normals.0
+                    && self.normals.2 == other.normals.1)
+        }
+    }
+
+    let mut unpacked_faces_1 = g_1.faces().iter().map(|face| match face {
+        Face::Triangle(f) => Some(UnpackedFace {
```
Comment (yanchith) [inline src/mesh_analysis.rs]: I think the `Some` wrapping is unnecessary, same below.

**Review by yanchith — CHANGES_REQUESTED** [review]
Mostly fine, I've just got a simple nit for the implementation and one clarification request for the doc comment.

File: src/mesh_analysis.rs (hunk 1 above)
Comment (janper) [reply src/mesh_analysis.rs]: no. a face with an opposite winding is a different face. "shifted winding" means (1, 2, 3) = (2, 3, 1) = (3, 1, 2) and that is still the same face

File: src/mesh_analysis.rs (hunk 2 above)
Comment (janper) [reply src/mesh_analysis.rs]: mind blown!

File: src/mesh_analysis.rs (hunk 1 above)
Comment (yanchith) [reply src/mesh_analysis.rs]: I understand, I just meant that if I didn't know it beforehand, the doc text wouldn't tell me :)

**Review by yanchith — APPROVED** [review]

**Pull request merged**